### PR TITLE
Automated cherry pick of #3337: Update prometheus operator permissions to v0.73.2

### PR DESCRIPTION
**Pull request by rene-dekker**
Cherry pick of #3337 on release-v1.34.

#3337: Update prometheus operator permissions to v0.73.2

### CODE REVIEW

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and delete the branch after I've merged it.

**Comment by marvin-tigera** [comment]
Removing "merge-when-ready" label due to new commits

**Review by vara2504 — APPROVED** [review]
lgtm

**Pull request merged**
